### DISCUSSION TITLE
Wave H2O - Fastapi integration

### DISCUSSION THREAD

**Discussion by mejimaru** [h2oai/wave#1739] [Q&A] [answered]
Hi everybody, 

Anybody know how to combine Wave H2O with Fastapi?

Wave Server <--> Wave Script <-->  Fastapi(Uvicorn server).

Thank you.

**Comment by mturoci** [comment, marked as answer]
Hi @mejimaru. FastAPI is meant for developing RESTful APIs while Wave is a UI framework.

We plan to make Wave embeddable into other popular python frameworks like Django, Flask, Tornado etc. next year. However, as of today, you will need to run Wave and FastAPI side-by-side as separate services. You can also have a look at [Nitro](https://nitro.h2o.ai/) which already allows this.

**Reply by mejimaru** [reply]
Thank you @mturoci, this H2O-Wave is life saver :) 

Is there any tutorial how to call external API into Wave? or just use httpx.get request?

**Reply by mturoci** [reply]
Glad you like it!

> Is there any tutorial how to call external API into Wave? or just use httpx.get request?

Everything is nothing else but Python so you are free to use any HTTP client you want. We usually use httpx -
https://github.com/h2oai/wave/blob/master/py/examples/http_client.py
